src/parser/parser.cpp

```diff
@@ -50,11 +50,22 @@ void Parser::DoNextStep() {
 }
 
 void Parser::ParseNextStep() {
+  // if there are parsing table errors do not perform parsing
+  if (!parsing_table_errors_.empty()) {
```

Comment:
this check is not required.

Reply:
Why parsing would not be performed if there is any errors in parsing_table
The input grammar should be correct!

Reply:
because the compiler already returns 0 in main. Having parsing_table_errors_ inside Parser class in bad design.

Reply:
Ahh Okay.

Reply:
Else do parsing_table_.GetErrors() or something having a extra variable is completely unnecessary.